src/main/scala/mesosphere/marathon/core/task/tracker/impl/TaskOpProcessorImpl.scala

```diff
@@ -0,0 +1,155 @@
+      * * an Action.Update if the tasks existed and the TaskStatus contains new information OR ELSE
+      */
+    def resolve(
+      appId: PathId, taskId: String, status: TaskStatus)(
+        implicit ec: ExecutionContext): Future[Action] = {
+      directTaskTracker.getTaskAsync(appId, taskId).map {
+        case Some(existingTask) =>
+          resolveForExistingTask(existingTask, status)
+        case None =>
+          Action.Fail(new IllegalStateException(s"task [$taskId] of app [$appId] does not exist"))
+      }
+    }
+
+    /**
+      * Calculates the change that needs to performed on this task according to the given task status update
+      */
+    private[this] def resolveForExistingTask(task: MarathonTask, status: TaskStatus): Action = {
+      def updateTaskOnStateChange(task: MarathonTask): Action = {
+        def statusDidChange(statusA: TaskStatus, statusB: TaskStatus): Boolean = {
+          val healthy = statusB.hasHealthy &&
+            (!statusA.hasHealthy || statusA.getHealthy != statusB.getHealthy)
+
+          healthy || statusA.getState != statusB.getState
+        }
+
+        if (statusDidChange(task.getStatus, status)) {
+          Action.Update(task.toBuilder.setStatus(status).build())
+        }
+        else {
+          log.debug(s"Ignoring status update for ${task.getId}. Status did not change.")
+          Action.Noop
+        }
+      }
+
+      status.getState match {
+        case TASK_ERROR | TASK_FAILED | TASK_FINISHED | TASK_KILLED | TASK_LOST =>
+          Action.Expunge
+        case TASK_RUNNING if !task.hasStartedAt => // was staged, is now running
+          Action.Update(task.toBuilder.setStartedAt(System.currentTimeMillis).setStatus(status).build())
+        case _ =>
+          updateTaskOnStateChange(task)
+      }
+    }
+
+  }
+}
+/**
+  * Processes durable operations on tasks by
+  *
+  * * storing the updated tasks in the task repository
+  * * informing the taskTracker actor of the latest task state
+  */
+private[tracker] class TaskOpProcessorImpl(
+    taskTrackerRef: ActorRef,
+    repo: TaskRepository,
+    statusUpdateActionResolver: StatusUpdateActionResolver) extends TaskOpProcessor {
+  private[this] val log = LoggerFactory.getLogger(getClass)
+
+  import TaskOpProcessor._
+
+  override def process(op: Operation)(implicit ec: ExecutionContext): Future[Unit] = {
+    op.action match {
+      case Action.Update(task) => // used for a create or as a result from a UpdateStatus action
+        repo.store(task).map { _ =>
+          taskTrackerRef ! TaskTrackerActor.TaskUpdated(op.appId, task, TaskTrackerActor.Ack(op.sender))
+        }.recoverWith(tryToRecover(op)(expectedTaskState = Some(task)))
+      case Action.Expunge => // used for task termination or as a result from a UpdateStatus action
+        repo.expunge(op.taskId).map { _ =>
+          taskTrackerRef ! TaskTrackerActor.TaskRemoved(op.appId, op.taskId, TaskTrackerActor.Ack(op.sender))
```

Comment:
The action has a sender. The sender is informed for Noop/Fail but not for update /Expunge/UpdateStatus.
Why?
Perhaps add this as comment?